### ISSUE TITLE
Enhancement: Allow OptionalPyblishPluginMixin to apply a tooltip to the checkbox

### ISSUE THREAD

**Issue by BigRoy** [ynput/OpenPype#5604]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

The `OptionalPyblishPluginMixin` makes it more trivial to implement optional checkboxes in the new publisher UI however, it'll always show the plug-in label as the checkbox label and generates no tooltip which can make it quite unclear what the checkbox does.

### How would you imagine the implementation of the feature?

Implement the following methods on the mixin:
```python
def get_optional_label(self):
    return self.label

def get_optional_tooltip(self):
    return None
```

These could then be overridden by child classes to e.g. provide a custom tooltip or label for the optional checkbox.

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

### Describe alternatives you've considered:

The other approach would be to not use the `OptionalPyblishPluginMixin` and then generate the attribute definitions manually so you can set the labels and tooltips as wanted.

**Comment by MustafaJafar** [comment]
Are you referring to this `active` toggle ? 
https://github.com/ynput/OpenPype/blob/6e574aff4d21d68df099a02b1c653a955a571615/openpype/pipeline/publish/publish_plugins.py#L199

So, should it be?
```python
class OptionalPyblishPluginMixin(OpenPypePyblishPluginMixin):

    def get_optional_label(self):
        return self.label
    
    def get_optional_tooltip(self):
        return None

    @classmethod
    def get_attribute_defs(cls):
        """Attribute definitions based on plugin's optional attribute."""

        # Empty list if plugin is not optional
        if not getattr(cls, "optional", None):
            return []

        # Get active value from class as default value
        active = getattr(cls, "active", True)
        # Return boolean stored under 'active' key with label of the class name
        label = cls.get_optional_label() or cls.__name__
        tooltip =  cls.get_optional_tooltip()
        return [
            BoolDef("active", default=active, label=label, tooltip=tooltip)
        ]
```
then developers in publish plugin should over ride them after making the secondary inheritance `OptionalPyblishPluginMixin` 
```
def get_optional_label(self):
    return "my label"

def get_optional_tooltip(self):
    return "my active toggle tool tip"
```

**Comment by BigRoy** [comment]
Yes, that's what I thought of as a design. Whether it's preferred that way I'll leave open to others :D
